src/Hazelcast.Net/Predicates/IPagingPredicate.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+
+namespace Hazelcast.Predicates
+{
+    /// <summary>
+    /// Defines a paging predicate.
+    /// </summary>
+    public interface IPagingPredicate : IPredicate
+    {
+        /// <summary>
+        /// Resets the predicate for re-use.
+        /// </summary>
+        void Reset();
+
+        /// <summary>
+        /// Moves to the next page.
+        /// </summary>
+        /// <remarks>
+        /// <para>This is equivalent to incrementing <see cref="Page"/>.</para>
+        /// </remarks>
+        void NextPage();
+
+        /// <summary>
+        /// Moves to the previous page.
+        /// </summary>
+        /// <remarks>
+        /// <para>This is equivalent to decrementing <see cref="Page"/>.</para>
+        /// </remarks>
+        void PreviousPage();
+
+        /// <summary>
+        /// Gets or sets the current page index
+        /// </summary>
+        int Page { get; set; }
+
+        /// <summary>
+        /// Gets the page size of each iteration.
+        /// </summary>
+        int PageSize { get; }
+
+        // FIXME comparer and anchor
+        // in Java they are typed ie KeyValuePair<TKey, TValue>, here it is <object, object>
+        // is this working as expected?
+        // is it OK not to be typed?
+        // note that being typed would require ... the query itself to be typed?
+
+        /// <summary>
+        /// Gets the comparer that is used to sort the results on the client.
+        /// </summary>
+        /// <remarks>
+        /// <para>The comparer implementation should be serializable by Hazelcast,
+        /// and should have a corresponding identical implementation on the server.</para>
+        /// </remarks>
+        IComparer<KeyValuePair<object, object>> Comparer { get; }
```

Comment:
I guess we can remove this method. For a client, this does not offer anything valuable. Maybe it makes sense for Java since server and the client are using the same interface

Reply:
Made it internal